src/shims/backtrace.rs

```diff
@@ -92,8 +92,18 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             throw_ub_format!("expected function pointer, found {:?}", ptr);
         };
 
-        if dest.layout.layout.fields.count() != 4 {
-            throw_ub_format!("bad declaration of miri_resolve_frame - should return a struct with 4 fields");
+        // Reconstruct the original function pointer,
+        // which we pass to user code.
+        let mut fn_ptr = ptr;
+        fn_ptr.offset = Size::from_bytes(0);
+        let fn_ptr = Scalar::Ptr(fn_ptr);
+
+        let num_fields = dest.layout.layout.fields.count();
+
+        if num_fields != 4 && num_fields != 5 {
+            // Always mention 5 fields, since the 4-field struct is only supported
+            // for backwards compatiblity. New code should declare 5 fields
```

Comment:
```suggestion
            // for backwards compatiblity. New code should declare 5 fields.
```

Reply:
But OTOH, given that we only just introduced this, is it even worth keeping compatibility with the 4-field case?

If we do keep compatibility, I think the docs should mention that, as it is confusing when docs and code diverge. If we consider the 4-field variant deprecated, I am in favor of just dropping support here and now, since it seems unlikely that anyone is already using this feature. This will break your libbacktrace integration but I assume you wanted to update that anyway with this change?

So basically, I think we should either officially support both 4 and 5 fields, or we should deprecate 4 fields and drop them either right now or within the next few weeks.

Reply:
I'm fine with dropping the 4-field support sometime after this is merged.